### PR TITLE
add a cache check and start running python LTC tests in CI

### PR DESCRIPTION
**Pull request by Krovatkin**
Fixes #ISSUE_NUMBER

### CODE REVIEW

**Comment by Krovatkin** [comment]
@malfet @janeyx99 we'd like to start running Lazy Tensor python tests in CI. Can I stick them in `extra_tests`? 

fyi: @wconstab

File: test/run_test.py [hunk 1]
```diff
@@ -134,6 +134,8 @@ def skip_test_p(name: str) -> bool:
         "distributed/elastic/utils/distributed_test",
         "distributed/elastic/multiprocessing/api_test",
         "test_deploy",
+        "lazy/test_misc.py",
```
Comment (janeyx99) [inline test/run_test.py:137]: @Krovatkin you shouldn't have to! These should be picked up already by default as our normal way of garnering tests is sweeping the test directory for files starting with `test_`

Comment (Krovatkin) [reply test/run_test.py:137]: this sounds awesome!

Comment (janeyx99) [reply test/run_test.py:137]: To verify, you can grep for `run_test.py` in the logs of any of the default CI jobs and check that test_misc.py is getting run in the selected tests list for one of the shards.

Comment (Krovatkin) [reply test/run_test.py:137]: @janeyx99 thanks! I expect this test to fail for now until another PR with the fix is checked in :) and it's failing alright : https://github.com/pytorch/pytorch/runs/6199967065?check_suite_focus=true

File: test/lazy/test_misc.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        def run_once(model, optimizer):
+            input = torch.ones(1, 1, 28, 28, device=device, requires_grad=True)
+            target = torch.ones(1, device=device, dtype=torch.int64)
+            optimizer.zero_grad(set_to_none=True)
+            output = model(input)
+            loss = F.nll_loss(output, target)
+            loss.backward()
+            optimizer.step()
+            torch._lazy.mark_step()
+            _ = loss.cpu()
```
Comment (wconstab) [inline test/lazy/test_misc.py:57]: what's the point of doing loss.cpu()?  is it possible that doing this triggers sync/compile and hides an issue if mark_step is broken?

File: test/lazy/test_misc.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+        for i in range(3):
+            run_once(model, optimizer)
+            self.assertEqual(None, torch._lazy.metrics.counter_value('UncachedCompile'))
```
Comment (wconstab) [inline test/lazy/test_misc.py:61]: actually i'm a bit surprised; wouldn't UncachedCompile == 1 after the first run, and then CachedCompile would == 2 after i=3?
